update(JS): web/javascript/reference/global_objects/array/index.md (#278)
webdoky/content: files/uk/web/javascript/reference/global_objects/array/index.md

  - : Повертає новий масив, утворений застосуванням переданої функції зворотного виклику до кожного елемента поточного масиву, а тоді – сплощенням результату на один рівень.
- {{jsxref("Array.prototype.forEach()")}} ("для кожного")

  - : Викликає функцію для кожного елемента поточного масиву.

- {{jsxref("Array.prototype.groupBy()")}} ("групувати за")

  - : Групує елементи масиву згідно з результатами перевіркової функції. До груп в результаті можна доступитись як до властивостей об‘єкта. Метод `groupBy()` групує елементи переданого масиву згідно зі значеннями, повернених перевірковою функцією. Повернений об‘єкт має окрему властивість для кожної групи, що містить масив елементів відповідної групи.

- {{jsxref("Array.prototype.includes()")}} ("включає")
  - : Визначає, чи містить поточний масив певне значення, повертаючи `true` чи `false` відповідно.
- {{jsxref("Array.prototype.indexOf()")}} ("індекс")
  - : Повертає перший (якнайменший) індекс, за яким переданий елемент може бути знайдений у поточному масиві.
- {{jsxref("Array.prototype.join()")}} ("об‘єднати")
  - : Об‘єднує усі елементи масиву в рядок.
- {{jsxref("Array.prototype.keys()")}} ("ключі")

>   - : Повертає новий [_ітератор масиву_](/uk/docs/Web/JavaScript/Guide/Iterators_and_Generators), що містить кожен індекс у поточному масиві.

Reviewer: ```suggestion
  - : Повертає новий [_ітератор масиву_](/uk/docs/Web/JavaScript/Guide/Iterators_and_Generators), що містить кожен індекс поточного масиву.
```

- {{jsxref("Array.prototype.map()")}} ("відобразити")
  - : Повертає новий масив, що містить результати виклику функції на кожному елементі поточного масиву.
- {{jsxref("Array.prototype.pop()")}} ("вистрелити")
  - : Прибирає останній елемент з масиву і повертає його.
- {{jsxref("Array.prototype.push()")}} ("запхати")
  - : Додає один чи більше елементів у кінець масиву і повертає нове значення `length` масиву.
- {{jsxref("Array.prototype.reduce()")}} ("звести")
  - : Виконує надану користувачем функцію-редуктор над кожним елементом масиву (зліва направо), аби звести його до єдиного значення.
- {{jsxref("Array.prototype.reduceRight()")}} ("звести справа")
  - : Виконує надану користувачем функцію-редуктор над кожним елементом масиву (справа наліво), аби звести його до єдиного значення.
- {{jsxref("Array.prototype.reverse()")}} ("перевернути")
  - : Розвертає порядок елементів масиву _на місці_. (Перший стає останнім, останній – першим.)
- {{jsxref("Array.prototype.shift()")}} ("зсунути")
  - : Прибирає перший елемент з масиву і повертає його.

> - {{jsxref("Array.prototype.slice()")}} ("вирізати")

Reviewer: Пропоную тримати ці підказки консистентними. https://github.com/webdoky/content/pull/273/.
Я б радше зупинився просто на зрізі, але можливо цей термін буде недостатньо зрозумілим, тому хай буде хоча б консистентно
```suggestion
- {{jsxref("Array.prototype.slice()")}} ("вирізати скибку, зріз")
```

```js
const fruits = ['Яблуко', 'Банан'];
const fruitsString = fruits.join(', ');
console.log(fruitsString);
// "Яблуко, Банан"
```

### Доступ до елемента масиву за його індексом

Цей приклад показує, як отримати елементи масиву `fruits` шляхом вказання числа, що є індексом їх позиції в масиві.

```js
const fruits = ['Яблуко', 'Банан'];

// Індекс першого елемента масиву – завжди 0.
fruits[0]; // Яблуко

// Індекс другого елемента масиву – завжди 1.
fruits[1]; // Банан

// Індекс останнього елемента масиву – завжди на один
// манший за довжину масиву.
fruits[fruits.length - 1]; // Банан

// Використання індексу, що більший за довжину масиву,
// повертає 'undefined'.
fruits[99]; // undefined
```

### Пошук індексу елемента в масиві

Цей приклад використовує метод [`indexOf()`](/uk/docs/Web/JavaScript/Reference/Global_Objects/Array/indexOf) для пошуку позиції (індексу) рядка "`Банан`" у масиві `fruits`.

```js
const fruits = ['Яблуко', 'Банан'];
console.log(fruits.indexOf('Банан'));
// 1
```

### Перевірка присутності певного елемента в масиві

> Цей приклад показує два способи перевірки того, чи містить масив `fruits` значення "`Банан`" і "`Вишня`": спершу за допомогою [`includes()`](/uk/docs/Web/JavaScript/Reference/Global_Objects/Array/includes), а тоді – із методом [`indexOf()`](/uk/docs/Web/JavaScript/Reference/Global_Objects/Array/indexOf), щоб перевірити, чи не рівний повернений індекс `-1`.

Reviewer: ```suggestion
Цей приклад показує два способи перевірки того, чи містить масив `fruits` значення "`Банан`" і "`Вишня`": спершу за допомогою [`includes()`](/uk/docs/Web/JavaScript/Reference/Global_Objects/Array/includes), а тоді – із методом [`indexOf()`](/uk/docs/Web/JavaScript/Reference/Global_Objects/Array/indexOf), щоб перевірити, чи повернений індекс не дорівнює `-1`.
```

```js
const fruits = ['Яблуко', 'Банан', 'Полуниця', 'Манго'];
const start = 1;
const deleteCount = 2;
const removedItems = fruits.splice(start, deleteCount);
console.log(fruits);
// ["Яблуко", "Манго"]
console.log(removedItems);
// ["Банан", "Полуниця"]
```

### Заміна кількох елементів масиву

Цей приклад використовує метод [`splice()`](/uk/docs/Web/JavaScript/Reference/Global_Objects/Array/splice) для заміни останніх 2 елементів масиву `fruits` новими елементами.

```js
const fruits = ['Яблуко', 'Банан', 'Полуниця'];
const start = -2;
const deleteCount = 2;
const removedItems = fruits.splice(start, deleteCount, 'Манго', 'Вишня');
console.log(fruits);
// ["Яблуко", "Манго", "Вишня"]
console.log(removedItems);
// ["Банан", "Полуниця"]
```

### Перебір масиву

Цей приклад використовує цикл [`for...of`](/uk/docs/Web/JavaScript/Reference/Statements/for...of) для перебору масиву `fruits`, виводячи кожний елемент у консоль.

```js
const fruits = ['Яблуко', 'Манго', 'Вишня'];
for (const fruit of fruits) {
  console.log(fruit);
}
// Яблуко
// Манго
// Вишня
```

> Утім, `for...of` – лише один з багатьох способів перебору будь-якого масиву; більше способів дивіться [Цикли та перебирання](/uk/docs/Web/JavaScript/Guide/Loops_and_iteration), а також документацію методів [`every()`](/uk/docs/Web/JavaScript/Reference/Global_Objects/Array/every), [`filter()`](/uk/docs/Web/JavaScript/Reference/Global_Objects/Array/filter), [`flatMap()`](/uk/docs/Web/JavaScript/Reference/Global_Objects/Array/flatMap), [`map()`](/uk/docs/Web/JavaScript/Reference/Global_Objects/Array/map), [`reduce()`](/uk/docs/Web/JavaScript/Reference/Global_Objects/Array/Reduce) і [`reduceRight()`](/uk/docs/Web/JavaScript/Reference/Global_Objects/Array/reduceRight) – ну і наступний приклад, що використовує метод [`forEach()`](/uk/docs/Web/JavaScript/Reference/Global_Objects/Array/forEach).

Reviewer: ```suggestion
Утім, `for...of` – лише один з багатьох способів перебору будь-якого масиву; більше способів можна знайти у розділі [Цикли та перебирання](/uk/docs/Web/JavaScript/Guide/Loops_and_iteration), а також у документації методів [`every()`](/uk/docs/Web/JavaScript/Reference/Global_Objects/Array/every), [`filter()`](/uk/docs/Web/JavaScript/Reference/Global_Objects/Array/filter), [`flatMap()`](/uk/docs/Web/JavaScript/Reference/Global_Objects/Array/flatMap), [`map()`](/uk/docs/Web/JavaScript/Reference/Global_Objects/Array/map), [`reduce()`](/uk/docs/Web/JavaScript/Reference/Global_Objects/Array/Reduce) і [`reduceRight()`](/uk/docs/Web/JavaScript/Reference/Global_Objects/Array/reduceRight) – ну і наступний приклад, що використовує метод [`forEach()`](/uk/docs/Web/JavaScript/Reference/Global_Objects/Array/forEach).
```
